source/auth/auth.rst

```diff
@@ -232,7 +232,11 @@ MONGODB-X509
 
 :since: 2.6
 
-MONGODB-X509 is the usage of X-509 certificates to validate a client.  The server will use the distinguished subject name of the client certificate in the SSL negotiation to authenticate. The driver will be required to supply the distinguished subject name outside of the SSL negotiation to the server using the "authenticate" command.
+MONGODB-X509 is the usage of X-509 certificates to validate a client.  The server will use the distinguished subject name of the client certificate in the SSL negotiation to authenticate. The driver MAY supply the distinguished subject name outside of the SSL negotiation to the server using the "authenticate" command.
+
+:since: 3.4
+
+MongoDB 3.4 no longer requires the user or the drivers to provide the username. Drivers that have the ability to automatically extract the distinguished subject name from a certificate MAY do so when no username is otherwise provided for backwards compatibility of earlier MongoDB versions. A driver SHOULD NOT preemptively error when no username is provided.
```

Comment:
The server doesn't say why authentication fails when it fails, it just says 'auth fails' irregardless if its missing username, invalid username, or wrong password.

So yes. If the user doesn't provide the driver with a username, and gets "auth fails", the driver does not know if thats because the user didn't provide us with username or if its because the certificate was rejected.

I don't see how it matters though. Do we really want to add extra version complexities if the user didn't provide us with username to check if he should have?

One of many problems the server resolved with the change was the fact users don't actually know their username when they are using X.509 auth. They just get issued certificates and thats the end of it.

The requirement of providing the username is now legacy, not the expected behavior. I think its sufficient to say "for MongoDB 3.4 compatible drivers, username is optional when authenticating over MONGODB-X509 when talking to MongoDB 3.4" ? :)